File: dmoj/ccc24s3/ccc24s3.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;
typedef pair<int, int> pii;
typedef vector<int> vi;
typedef vector<pii> vpii;
#define pb push_back
#define fs first
#define sn second
#define ms(a, x) memset(a, x, sizeof(a))
#define ALL(v) begin(v), end(v)
#define SZ(v) ((int) (v).size())
const int INF = 0x3f3f3f3f;
#define FR(i, n) for(int i = 0; i < (n); i++)
#define FOR(i, a, b) for(int i = (a); i < (b); i++)
#define FORR(i, a, b) for(int i = (a); i >= (b); i--)
#define dbg(x){cerr << #x << ' ' << (x) << endl;}
#define dbgArr(arr, n){cerr << #arr; FR(_i, n) cerr << ' ' << (arr)[_i]; cerr << endl;}
template <typename T, typename U>
ostream& operator<<(ostream &os, pair<T, U> p){return os << "(" << p.fs << ", " << p.sn << ")";}

const int MN = 3e5 + 5;
int n, m;
int arr[MN], brr[MN], crr[MN];

int main(){
    cin.sync_with_stdio(0); cin.tie(0);
    cin >> n;
    FR(i, n) cin >> arr[i];
    FR(i, n) cin >> brr[i];
    FR(i, n) crr[i] = brr[i];
    m = unique(crr, crr+n) - crr;

    vi aVec;

    int ind = 0;
    FR(i, n){
        if(ind < m && arr[i] == crr[ind]){
            ind++;
            aVec.pb(i);
        }
    }

    if(ind != m) return !printf("NO\n");
    cout << "YES\n";

    int lst = -1;
    vpii bVec;
    FR(i, n){
        if(bVec.empty() || brr[i] != lst) bVec.pb({i, i});
        else bVec.back().sn = i;
        lst = brr[i];
    }
    assert(SZ(bVec) == m);

    vector<pair<char, pii>> ans;
    FR(i, m){
        int p = aVec[i];
        auto [l, r] = bVec[i];
        if(p > l) ans.pb({'L', {l, p}});
    }
    FORR(i, m-1, 0){
        int p = aVec[i];
        auto [l, r] = bVec[i];
        if(p < r) ans.pb({'R', {p, r}});
    }

    int k = SZ(ans);
    cout << k << '\n';
    for(auto [ch, p] : ans) cout << ch << ' ' << p.fs << ' ' << p.sn << '\n';
}

File: dmoj/ccc24s3/ccc24s3.py
import sys

n = int(input())
arr = list(map(int, input().split()))
brr = list(map(int, input().split()))

blks = [[0, 0]]
for i in range(n):
    if brr[i] == brr[blks[-1][0]]:
        blks[-1][1] = i
    else :
        blks.append([i, i])

m = len(blks)
a_pos = []
blk_ind = 0

for i in range(n):
    if blk_ind < m and arr[i] == brr[blks[blk_ind][0]]:
        a_pos.append(i)
        blk_ind += 1

if blk_ind != len(blks):
    print('NO')
    sys.exit()

ans = []

for i in range(m):
    p = a_pos[i]
    l, r = blks[i]
    if p > l:
        ans.append(('L', l, p))

for i in range(m-1, -1, -1):
    p = a_pos[i]
    l, r = blks[i]
    if p < r:
        ans.append(('R', p, r))

print('YES')
print(len(ans))

for ch, l, r in ans:
    print(ch, l, r)
